packages/flutter/lib/src/rendering/box.dart

```diff
@@ -90,7 +90,10 @@ class BoxConstraints extends Constraints {
     this.maxWidth = double.infinity,
     this.minHeight = 0.0,
     this.maxHeight = double.infinity,
-  });
+  }) : assert(minWidth != null),
```

Comment:
As noted earlier: this change, along with a test that checks the assertion, should be landed first, in a separate PR.
```
    expect(() => BoxConstraints(minWidth: null), throwsAssertionError);
    expect(() => BoxConstraints(maxWidth: null), throwsAssertionError);
    ...
```

Reply:
Resolved in #48295.